elixir/lib/contracts.ex

```diff
@@ -60,10 +64,10 @@ defmodule Trento.Contracts do
   Decode and unwrap a protobuf CloudEvent to an event struct.
   """
   @spec from_event(binary(), Keyword.t()) ::
-          {:ok, struct()}
+          {:ok, struct(), map()}
```

Comment:
I would try to not change the signature here.
Maybe we could have a new function like:
```
@spec attributes_from_event(binary(), Keyword.t()) :: {:ok, map()} | {:error, any}
```

Reply:
This is also an option I considered.

To understand better, `attributes_from_event` would get an encoded cloud event, similarly to what `from_event` receives and return attributes.

Meaning that when we need both the decoded event and the attributes we'd be calling both `Contracts.from_event(cloud_event)` and `Contracts.attributes_from_event(cloud_event)`.

Asking not because I am against it, but just to get it right 😄

Reply:
Exactly. We don't need the attributes in most of the cases, so we could just use that function in the cases where we need them.

Reply:
I am fine with it, it would mean decoding twice in those cases, but it's ok